code/modules/reagents/chemistry/reagents/drugs.dm

```diff
@@ -369,31 +427,37 @@
 	addiction_chance = 15
 	addiction_threshold = 5
 	metabolization_rate = 0.6
+	addiction_decay_rate = 0.2
 	taste_description = "WAAAAGH"
+	/// timer until we can start rolling for reducing a mobs strength again.
+	var/next_remove_strength
+
+/datum/reagent/bath_salts/on_mob_add(mob/living/L)
+	if(ishuman(L))
+		var/mob/living/carbon/human/H = L
+		H.physiology.melee_bonus += 2 // rage mode
 
 /datum/reagent/bath_salts/on_mob_life(mob/living/M)
 	var/check = rand(0,100)
 	var/update_flags = STATUS_UPDATE_NONE
 	if(ishuman(M))
 		var/mob/living/carbon/human/H = M
-		var/obj/item/organ/external/head/head_organ = H.get_organ("head")
-		if(check < 8 && head_organ.h_style != "Very Long Beard")
-			head_organ.h_style = "Very Long Hair"
-			head_organ.f_style = "Very Long Beard"
-			H.update_hair()
-			H.update_fhair()
-			H.visible_message("<span class='warning'>[H] has a wild look in [H.p_their()] eyes!</span>")
-	if(check < 60)
-		M.SetParalysis(0)
-		M.SetStunned(0)
-		M.SetWeakened(0)
+		var/recent_consumption = holder.addiction_threshold_accumulated[type]
+		for(var/obj/item/organ/internal/I in H.internal_organs)
+			I.receive_damage(0.8, TRUE) //double the rate of mitocholide
+		if(world.time > next_remove_strength && recent_consumption > 5 && prob(0.1 * DRAWBACK_CHANCE_MODIFIER(recent_consumption))) // tiny chance to make their muscles waste away, cannot happen instantly. I don't want people to get *that* unlucky
+			H.physiology.melee_bonus--
+			to_chat(H, "<span class='biggerdanger'>You feel your muscles wasting away!</span>")
+			next_remove_strength = world.time + 100 SECONDS
+	M.SetParalysis(0)
+	M.SetStunned(0)
+	M.SetWeakened(0)
+	M.Druggy(30 SECONDS)
```

Comment:
Same here, for each second on bath salts, you'd spend 28 seconds being high after no bath salts are in you.
But this one is pre-existing, so it's probably fine?..

Reply:
This isnt an adjust proc, so it doesn't go above 30 seconds.